### PR TITLE
KRKNWK-8785: Create sdk-802154

### PR DESCRIPTION
**Pull request by JakubPegza**
This repository contains new structure of sdk-802154(TBD) repository. Along with it changes in:
https://github.com/JakubPegza/zephyr/tree/feature/add_sdk_802154_repo
https://github.com/JakubPegza/sdk-nrfxlib/tree/feature/remove_nrf_802154_sl
are required

### CODE REVIEW

File: serialization/src/include/nrf_802154_serialization_crit_sect.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+/**
+ * @file nrf_802154_serialization_crit_sect.h
```
Comment (AdamZelikNS) [inline serialization/src/include/nrf_802154_serialization_crit_sect.h:36]: I found this file to be duplicate. There is `serialization/include/platform/nrf_802154_serialization_crit_sect.h`.
It looks like it needs to be removed from here, as well as a few others:
`serialization/src/include/nrf_802154_spinel_backend.h`
`serialization/src/include/nrf_802154_spinel_backend_callouts.h`
`serialization/src/include/nrf_802154_spinel_log.h`
`serialization/src/include/nrf_802154_spinel_response_notifier.h`

**Comment by AdamZelikNS** [comment]
The serialization platform headers are here in 
`serialization/include/platform/`
This is different from what was finally adopted in [PR#1](https://github.com/pkwiek/hal-nrf-802154/pull/1) into the `hal-nrf-802154` repo, where it looks like this:
`serialization/include/serialization/platform/`
Is this difference intentional?

**Review by AdamZelikNS — CHANGES_REQUESTED** [review]

**Comment by e-rk** [comment]
Are the bare metal examples going to be included in this PR or are they going to be a part of some stage 2 PR?

**Comment by JakubPegza** [comment]
> The serialization platform headers are here in
> `serialization/include/platform/`
> This is different from what was finally adopted in [PR#1](https://github.com/pkwiek/hal-nrf-802154/pull/1) into the `hal-nrf-802154` repo, where it looks like this:
> `serialization/include/serialization/platform/`
> Is this difference intentional?

Not intentional, but already aligned between these two repositories. Good catch though

**Comment by JakubPegza** [comment]
> Are the bare metal examples going to be included in this PR or are they going to be a part of some stage 2 PR?

They are missing here yet because I think they were still not merged when I branched off. They will end up in this PR

**Comment by jciupis** [comment]
Is it OK to close this PR?

**Comment by JakubPegza** [comment]
Outdated solution, closing

**Pull request closed without merging**